cli/src/main/java/de/jplag/cli/logger/TerminalPrinter.java

```diff
@@ -0,0 +1,87 @@
+package de.jplag.cli.logger;
+
+import java.io.PrintStream;
+import java.util.PriorityQueue;
+import java.util.Queue;
+
+/**
+ * Prints strings to stdout. Provides the option to delay the actual printing.
+ */
+public class TerminalPrinter {
+    private static TerminalPrinter instance;
+
+    /**
+     * Threadsafe singleton getter
+     * @return The singleton instance
+     */
+    public static TerminalPrinter getInstance() {
+        synchronized (TerminalPrinter.class) {
+            if (instance == null) {
+                synchronized (TerminalPrinter.class) {
+                    instance = new TerminalPrinter();
+                }
+            }
+        }
+
+        return instance;
+    }
```

Comment:
I would either synchronize the whole method (draconian synchronization) or do the double-checked locking pattern here instead of synchronizing twice.

Reply:
Interesting, did it give you input as to why it wants this implementation?